editor/filesystem_dock.cpp

```diff
@@ -2556,6 +2677,29 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 			String dir = ProjectSettings::get_singleton()->globalize_path(fpath);
 			ScriptEditor::get_singleton()->open_text_file_create_dialog(dir);
 		} break;
+		default: {
+			// Resource conversion commands:
+			if (p_option >= CONVERT_BASE_ID) {
+				HashSet<String> valid_conversion_targets = _get_valid_conversions_for_file_paths(p_selected);
+				selected_conversion_id = p_option - CONVERT_BASE_ID;
+				ERR_FAIL_COND(selected_conversion_id >= (int)valid_conversion_targets.size());
+
+				to_convert.clear();
+				for (const String &S : p_selected) {
+					to_convert.push_back(S);
+				}
+
+				int conversion_id = 0;
+				for (const String &E : valid_conversion_targets) {
+					if (conversion_id == selected_conversion_id) {
+						conversion_dialog->set_text(vformat(TTR("Do you wish to convert these files to %s? (This operation cannot be undone!)"), E));
```

Comment:
The conversion dialog has no other text, so you can set it in the constructor instead of every time.

Reply:
I'm a bit confused about this since the text set here is formatted to include the name of the resource type you're converting to. Can you clarify what you mean?

Reply:
I mean that I missed vformat, sorry.